app/src/main/java/com/hedvig/app/feature/offer/usecase/GetQuoteIdsUseCase.kt

```diff
@@ -0,0 +1,30 @@
+package com.hedvig.app.feature.offer.usecase
+
+import com.hedvig.app.feature.offer.OfferRepository
+import com.hedvig.app.util.apollo.QueryResult
+import com.hedvig.app.util.apollo.safeQuery
+
+class GetQuoteIdsUseCase(
+    private val offerRepository: OfferRepository,
+) {
+    sealed class Result {
+        data class Success(
+            val ids: List<String>,
+        ) : Result()
+
+        object Error : Result()
+    }
+
+    suspend operator fun invoke(ids: List<String>): Result {
```

Comment:
Nice! Could we move this request from offerRepository to this useCase as well? 
```
fun offer(ids: List<String>) = apolloClient
        .query(offerQuery(ids))
        .watcher()
        .toFlow()   
```

And then use the result with this function in the viewmodel (passing our Result instead of a Response)
```
    private fun toViewState(response: Response<OfferQuery.Data>): ViewState {
        return response.errors?.let {
            ViewState.Error.GeneralError(it.firstOrNull()?.message)
        } ?: response.data?.let { data ->
            if (data.contracts.isNotEmpty()) {
                ViewState.HasContracts
            } else {
                val items = OfferItemsBuilder.createItems(data)
                ViewState.OfferItems(items)
            }
        } ?: ViewState.Error.EmptyResponse
    }
```

Then we would only need to make changes in this class when removing support for `lastQuoteOfMember`.